Enhance color picker tool and remove GHLiveSample macro

OnkarRuikar opened the pull request:
The PR does following changes:
1. Uses livesample macro instead of GHLivesample macro by embedding the code in the content file
2. Adds more color spaces in the output color table 
3. Adds text color input field. This will help users convert their custom color  in various formats and color spaces.
4. Also, makes small tweeks to the other color mixer tool. 

The new look of the tool:

![5](https://github.com/user-attachments/assets/f74b2c52-dc37-46a0-9cc3-cba19a51818e)

OnkarRuikar committed:
Enhance color picker tool and remove GHLiveSample macro

dipikabh commented on files/en-us/web/css/css_colors/color_picker/index.md:
Also follow the style defined in https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Code_style_guide/CSS#colors

dipikabh commented on files/en-us/web/css/css_colors/color_picker/index.md:
```diff
@@ -15,7 +15,461 @@ This tool lets you pick a color in the sRGB {{glossary("color space")}} and conv
+      <caption>
+        Current color values:
+      </caption>
+      <tbody>
```
In both color picker and mixer tools, can we use this order (and casing) for the color formats:

RGB
HEX
HSL
HWB
color()
LAB
LCH
Oklab
Oklch
xyz-d50
xyz-d65

OnkarRuikar replied:
I've changed the orders.

The color space names casings `Lab`, `OkLCh`, `XYZ D50`, and `XYZ D65` have been taken from the specs:
> ![1](https://github.com/user-attachments/assets/67216ffd-d8d1-46a0-b752-4b0eb32cbd1f)

 - [Relative OkLCh Colors](https://drafts.csswg.org/css-color-5/#relative-OkLCh)
 - [The Predefined CIE XYZ Color Spaces](https://drafts.csswg.org/css-color-4/#predefined-xyz)
 - [D50](https://drafts.csswg.org/css-color-4/#d50)

It is better to use English names in the left column than CSS keywords, otherwise we can simply remove the left column, right?

dipikabh replied:
Thanks for updating the order.

Regarding the labels, I was referring to how we mention them in our docs :), specifically here: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#syntax. We need to be consistent across MDN.

Linking them all is a very nice touch! 🙌  Thanks

dipikabh reviewed (changes requested):
Thanks a lot, Onkar! This is a great contribution.

Are we also planning to delete the code from `mdn/css-examples`?

I'll also copy @estelle in case she has anything to add.

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

OnkarRuikar committed:
address review comments

OnkarRuikar commented:
> Are we also planning to delete the code from `mdn/css-examples`?

I haven't seen explicit deletions after copying code from other repos. Would be better if we keep those repos intact and archive them after all the stuff is copied.

Here is how it looks after addressing the review comments:

![1](https://github.com/user-attachments/assets/e4d01dc9-561f-477a-ab75-f695ab1fcb7b)

dipikabh commented on files/en-us/web/css/css_colors/color_picker/index.md:
```diff
@@ -5,19 +5,530 @@ page-type: guide
+        </tr>
+      </tbody>
+    </table>
+    <p>
+      <label for="color-text">Enter a color value in any format:</label><br />
+      <input type="text" id="color-text" /><input
+        type="color"
+        id="color-input" />
+    </p>
```
- The screenshot you've shared in https://github.com/mdn/content/pull/41158#issuecomment-3305202894 seems to have a color wheel image right after the text field. But I don't see it in the preview URL. There's also no `<img>` tag in code.
- It's not very clear from the UI that readers have two options - entering a color value or selecting one using the color picker.
  - Could we change the label so that it covers both: "Enter a color value in any format or pick a color:"
  - It just struck me that maybe we should call the tool "Color format converter"?
- Some spacing between the text field and the color input field might be nice.
- I think all the input interactions should appear at the top, followed by the table of generated values.
- Could we also center-align all the input fields?

dipikabh commented on files/en-us/web/css/css_colors/color_picker/index.md:
```diff
@@ -5,19 +5,530 @@ page-type: guide
+  <dialog open>
+    <table id="output-colors">
+      <caption>
+        Current color values:
```
or this?
```suggestion
        Your color in different formats:
```
OR
```suggestion
        Value of your color in different formats:
```
OR
```suggestion
        Your color equivalent in other formats:
```

dipikabh reviewed (changes requested):
Hi @OnkarRuikar, I have some more suggestions based on your latest changes.

dipikabh commented on files/en-us/web/css/css_colors/color_mixer/index.md:
L8:
- could you make the sentence starting with "The two input colors..." a new para
- drop the "on": "Click the outer swatches to select new colors to mix." 
- last sentence can be (also adding the copy feature): "The resulting color's values in different formats appear in a table following the widget and can be copied."

dipikabh replied:
L13: maybe we can get rid of this since we're providing the instructions in prose. It'll also help to reduce the amount of text and clutter in the UI.
>  <p>Click on 'color-one' and 'color-two' to change the input colors.</p>

dipikabh commented on files/en-us/web/css/css_colors/color_mixer/index.md:
```diff
@@ -105,48 +105,114 @@ This tool lets you mix two colors in any color space using the {{cssxref("color_
     The output color in various CSS color syntaxes:
```
"Mixed color in..." the rest can be similar to what we end up using in color picker (https://github.com/mdn/content/pull/41158#discussion_r2360387164)

dipikabh reviewed (changes requested):
some comments for the color mixer page

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

OnkarRuikar committed:
address review comments

OnkarRuikar committed:
correct colorspace name casings

OnkarRuikar commented:
> The screenshot you've shared in https://github.com/mdn/content/pull/41158#issuecomment-3305202894 seems to have a color wheel image right after the text field. But I don't see it in the preview URL. There's also no <img> tag in code.

My bad. It's the color picker browser extension that embeds the wheels. Here is how it looks after the review comments:
![1](https://github.com/user-attachments/assets/e646e1df-ad65-4092-804e-0702256afbd7)

> It just struck me that maybe we should call the tool "Color format converter"?

Do you mean change file name and slug to `color_format_converter` as well?

> Regarding the labels, I was referring to how we mention them in our docs :), specifically here: [developer.mozilla.org/en-US/docs/Web/CSS/color_value#syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#syntax). We need to be consistent across MDN.

Not all docs follow the casings used in `color_value` page. I am updating  correct casing  everywhere.

dipikabh reviewed (changes requested):
Thanks! This is close to ready. Just a few minor adjustments needed.

OnkarRuikar committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

OnkarRuikar committed:
review comments

OnkarRuikar commented on files/en-us/web/css/css_colors/color_mixer/index.md:
````diff
@@ -826,7 +886,7 @@ function init() {
 ```
 
-{{EmbedLiveSample("color-mixer", "", 850, "", "", "", "clipboard-write")}}
+{{EmbedLiveSample("color-mixer", "", 850, "", "", "", "clipboard-write", "allow-popups")}}
````
Now I've set it to 760. Firefox on Linux Mint consumes all the height and with same height Chrome leaves 2-3 rems empty at the bottom. So we need some extra height as buffer to avoid vertical scrollbar.

OnkarRuikar committed:
align inputs

OnkarRuikar committed:
better formatting

OnkarRuikar committed:
rename to color_format_converter

dipikabh commented on files/en-us/web/css/css_colors/color_format_converter/index.md:
```diff
@@ -0,0 +1,570 @@
+sidebar: cssref
+---
+
+This tool lets you enter or pick a color and copy its corresponding value in any CSS [color format](/en-US/docs/Web/CSS/color_value). You can use the generated color value anywhere the {{cssxref("color_value", "&lt;color&gt;")}} data type is supported in CSS. This tool can also help you understand the syntax of the different color notations – click any color format to learn more about it.
```
```suggestion
This tool lets you enter or pick a color and copy its corresponding value in any CSS [color format](/en-US/docs/Web/CSS/color_value). You can use the generated color value anywhere the {{cssxref("color_value", "&lt;color&gt;")}} data type is supported in CSS. This tool can also help you understand the syntax of the different color notations.
```

dipikabh committed:
remove text to avoid repitition

dipikabh reviewed (approved):
Great work, @OnkarRuikar 🙌 Thanks a lot!
